### PR TITLE
Issue #1054 - Ported the use-isnan rule from tslint-microsoft-contrib…

### LINKED ISSUE #1054: New rule: Equals signs can't be used to check if variable is NaN (use isNaN() instead)
Equals signs can't be used to check if variable is `NaN`, `isNaN()` function should be used instead.

#### Currently following incorrect usage doesn't raise an error

Example of **incorrect** usage:
```ts
const maybeNumber = parseInt("a");
if( maybeNumber === NaN || maybeNumber === NaN ) {
  // this line is never reached <- UNEXPECTED
}
```
#### Example of correct usage
```ts
const maybeNumber = parseInt("a");
if( isNaN(maybeNumber) ) {
  // this line is reached, as expected
}
```

### PR DESCRIPTION
**Pull request by HamletDRC**
… to tslint

Closes #1054

### CODE REVIEW

**Comment by HamletDRC** [comment]
It's not clear to me where to update your documentation pages. Do I need to do that as well?

**Comment by JKillian** [comment]
Don't worry about the documentation pages - I normally update those once a release with the changes gets published.

File: src/rules/useIsnanRule.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as ts from "typescript";
+import * as Lint from "../lint";
+
+/**
```
Comment (JKillian) [inline src/rules/useIsnanRule.ts]: Can delete this comment

File: src/rules/useIsnanRule.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+export class Rule extends Lint.Rules.AbstractRule {
+    public static FAILURE_STRING = "Found an invalid comparison for NaN: ";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new UseIsnanRuleWalker(sourceFile, this.getOptions()));
+    }
+}
+
+class UseIsnanRuleWalker extends Lint.RuleWalker {
+
```
Comment (JKillian) [inline src/rules/useIsnanRule.ts]: blank line is uneeded

File: test/tsconfig.json [hunk 3]
```diff
@@ -142,4 +143,4 @@
         "rule-tester/testData.ts",
         "rule-tester/utilsTests.ts"
     ]
-}
+}
```
Comment (JKillian) [inline test/tsconfig.json]: Can you re-add the newline at the end of this file? (So we prevent merge conflicts and unnecessary diffs)

**Comment by JKillian** [comment]
Thanks @HamletDRC, this PR looks great! Could you just make the super minor tweaks mentioned above before it merges?

File: src/rules/useIsnanRule.ts (hunk 2 above)
Comment (HamletDRC) [inline src/rules/useIsnanRule.ts]: Which blank line is that? This looks properly spaced to me.

File: src/rules/useIsnanRule.ts (hunk 1 above)
Comment (HamletDRC) [inline src/rules/useIsnanRule.ts]: Do you mean the block comment and copyright notice? Or the JSDoc?

File: test/tsconfig.json (hunk 3 above)
Comment (HamletDRC) [inline test/tsconfig.json]: Sure. However, this was added automatically by the grunt build. So you may want to change the build somehow. I am on Windows using git-bash. My .git/config file has: 

    core.autocrlf=input
    core.eol=lf

File: src/rules/useIsnanRule.ts (hunk 1 above)
Comment (JKillian) [inline src/rules/useIsnanRule.ts]: I meant the JSDoc, but actually it doesn't hurt, you can just leave it 👍

File: test/tsconfig.json (hunk 3 above)
Comment (JKillian) [inline test/tsconfig.json]: Oh, just ignore my previous comment in that case, it seems as if `grunt-ts` always removes this when rewriting `tsconfig.json` files. We might as well remove it and leave it removed.

File: src/rules/useIsnanRule.ts (hunk 2 above)
Comment (JKillian) [inline src/rules/useIsnanRule.ts]: Line 33, but no matter, not important enough at all to make you revise for. Merged!

**Comment by HamletDRC** [comment]
Great. Thanks.

**Comment by dasa** [comment]
This is now failing for:

```ts
    let x = 0;
    x = NaN;
```

**Comment by JKillian** [comment]
Uh-oh! Thanks for the heads-up @dasa

**Comment by HamletDRC** [comment]
taking a look

**Comment by HamletDRC** [comment]
sending pull request

**Pull request merged**